enterprise/forms.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""
+User-facing forms for the Enterprise app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from six.moves.urllib.parse import urlparse  # pylint: disable=import-error
+
+from django import forms
+from django.urls import Resolver404, resolve
+from django.utils.translation import ugettext as _
+
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser
+
+ENTERPRISE_SELECT_SUBTITLE = _(
+    u'You have access to multiple organizations. Select an organization '
+    'to view the courses associated with that organization. If you want to access a different organization '
+    'later, you will need to sign out and sign back in.'
+)
+
+
+class EnterpriseSelectionForm(forms.Form):
+    """
+    Enterprise Selection Form.
+    """
+
+    enterprise = forms.ChoiceField(choices=(), label='Organization')
+    success_url = forms.CharField(widget=forms.HiddenInput(), required=False)
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize form.
+        """
+        super(EnterpriseSelectionForm, self).__init__(*args, **kwargs)
+        initial = kwargs['initial']
+        self._user_id = kwargs['initial'].pop('user_id')
+        self.fields['enterprise'].choices = initial['enterprises']
+        self.fields['success_url'].initial = initial['success_url']
+
+    def clean(self):
+        """
+        Validate POST data.
+        """
+        cleaned_data = super(EnterpriseSelectionForm, self).clean()
+        enterprise = cleaned_data.get('enterprise')
+
+        try:
+            EnterpriseCustomer.objects.get(uuid=enterprise)  # pylint: disable=no-member
+        except EnterpriseCustomer.DoesNotExist:
+            raise forms.ValidationError(_("Enterprise not found"))
+
+        # verify that learner is really a member of selected enterprise
+        if not EnterpriseCustomerUser.objects.filter(enterprise_customer=enterprise, user_id=self._user_id).exists():
+            raise forms.ValidationError(_("Wrong Enterprise"))
+
+        try:
+            # urlparse returns ParseResult(scheme='', netloc='', path=u'', params='', query='', fragment='')
+            # we are validating if path is a valid url
+            resolve(urlparse(cleaned_data['success_url'])[2])
```

Comment:
I think `resolve(cleaned_data['success_url'])` should work we don't need to use `urlparse`

Reply:
@ziafazal I tried `resolve` on complete url and it gives `Resolver404` exception. Below is what I tried on sandbox django shell

```python
>>> from six.moves.urllib.parse import urlparse
>>> from django.urls import Resolver404, resolve
>>> url = 'https://mammar.sandbox.edx.org/enterprise/select/active'
>>> resolve(url)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/urls/base.py", line 27, in resolve
    return get_resolver(urlconf).resolve(path)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/urls/resolvers.py", line 395, in resolve
    raise Resolver404({'path': path})
Resolver404: {u'path': u'https://mammar.sandbox.edx.org/enterprise/select/active'}
>>> urlparse(url)
ParseResult(scheme='https', netloc='mammar.sandbox.edx.org', path='/enterprise/select/active', params='', query='', fragment='')
>>> resolve(urlparse(url)[2])
ResolverMatch(func=enterprise.views.EnterpriseSelectionView, args=(), kwargs={}, url_name=enterprise_select_active, app_names=[], namespaces=[])
```

Reply:
I thought we'd have relative url in `cleaned_data['success_url']`

Reply:
I have refactored the code and now success_url is a relative url.